### ISSUE TITLE
not seeing a sync changes button

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#153546]
Testing #153317

I see a sync & commit action in the button menu though

**Comment by lszomoru** [comment]
@meganrogge, would you be able to provide a screenshot? Thanks!

**Comment by meganrogge** [comment]
<img width="1471" alt="Screen Shot 2022-06-29 at 10 14 03 AM" src="https://user-images.githubusercontent.com/29464607/176458569-10de6f3e-12ca-43bd-84e6-8d95f97ce086.png">

**Comment by lszomoru** [comment]
@meganrogge, based on the screenshot, I see that you have one file that is currently being staged. In case there are any files that are either staged, or changed, we will show the "Commit" action button instead of the "Sync Changes" action button. Maybe there was a misunderstanding in the test plan steps?

**Comment by meganrogge** [comment]
yeah maybe I'm misunderstanding - how can I get the `Sync changes` action to show up?

**Comment by meganrogge** [comment]
ok I see it, but that's a bit unexpected bc I switched to a branch and pulled and now see the button but there shouldn't be any changes to sync.
<img width="465" alt="Screen Shot 2022-07-01 at 11 31 47 AM" src="https://user-images.githubusercontent.com/29464607/176925080-b0e711b0-93f9-48b4-bb35-9f346fc64e34.png">
when I clicked sync changes, this happened:

```
> git pull --tags origin main
From https://github.com/microsoft/vscode
 * branch                    main       -> FETCH_HEAD
 ! [rejected]                1.64.1     -> 1.64.1  (would clobber existing tag)
```

**Comment by lszomoru** [comment]
That is expected. During testing we have heard feedback that it is confusing that even if you have all action button settings enabled there were some scenarios in which no button was shown. To address that feedback, we have decided that in case the branch has an upstream, and there are no changes, we will *always* show the `Sync Changes` action button as depending on your user settings, you might not have the latest changes from the remote branch. Happy to discuss further if you want.

**Comment by meganrogge** [comment]
> depending on your user settings

Can we only show it _always_ if the settings are such that you might not have the latest changes?

**Comment by lszomoru** [comment]
I think that that is something that we can consider.

**Comment by ss-sec** [comment]
Please fix the bug "Syn Changes" button being always enabled - It is bit confusing.  Before it was only if there is any changes that need to updated - It was more straightforward.

**Comment by meganrogge** [comment]
still not seeing the sync changes button (just commit) when I've staged changes. how should this be verified @lszomoru ?

**Comment by lszomoru** [comment]
@meganrogge, at the moment you would only see the "Sync Changes" button if you do not have any pending changes, *and* we are certain that your local branch is either ahead/behind the remote branch. After clicking "Sync Changes" you should see a disabled "Commit" button.
